proof/crefine/ARM/ArchMove_C.thy

```diff
@@ -235,6 +235,22 @@ crunches insertNewCap, Arch_createNewCaps, threadSet, Arch.createObject, setThre
    simp: unless_def updateObject_default_def crunch_simps
    ignore_del: preemptionPoint)
 
+lemma addrFromPPtr_mask[simplified ARM.pageBitsForSize_simps]:
+  "n \<le> pageBitsForSize ARMSuperSection
+   \<Longrightarrow> addrFromPPtr ptr && mask n = ptr && mask n"
+  apply (simp add: addrFromPPtr_def)
+  apply (prop_tac "pptrBaseOffset AND mask n = 0")
+   apply (rule mask_zero[OF is_aligned_weaken[OF pptrBaseOffset_aligned]], simp)
+  apply (simp flip: mask_eqs(8))
+  done
+
+(* FIXME MK: this can be done as:
+   lemmas addrFromPPtr_mask_5 = addrFromPPtr_mask[where n=5, simplified]
+   but that doesn't give a sanity check of the n \<le> ... assumption  disappearing *)
+lemma addrFromPPtr_mask_5:
+  "addrFromPPtr ptr && mask 5 = ptr && mask 5"
+  by (rule addrFromPPtr_mask[where n=5, simplified])
```

Comment:
I think it's Ok to drop the fixme and just leave the rest of the comment there as explanation why it is done this way.

Reply:
It would be a bit over the top, but if we want to enforce that the assumption disappears without explicitly writing out the lemma we could do something like https://github.com/seL4/l4v/blob/rt/proof/refine/ARM/Invariants_H.thy#L3013